packages/grafana-schema/src/schema/mudball.gen.ts

```diff
@@ -473,12 +480,40 @@ export enum BarGaugeDisplayMode {
   Lcd = 'lcd',
 }
 
+/**
+ * Table cell options. Each cell has a display mode
+ * along with sub options for that display mode. Sub options
+ * are grouped within the subOptions property by display mode.
+ */
+export interface TableCellOptions {
+  displayMode: TableCellDisplayMode;
+  subOptions: TableCellSubOptionGroup;
+}
```

Comment:
I still don't think this is the right way to type this. This should be a union type with a discriminator enum 

subOptions is not need

Reply:
I've been thinking about this and I think the downside of using a union type with discriminator is that configuration for cell types that have been previously configured can be lost. Seemed like a somewhat jarring experience. For instance, setting a colored background to "gradient" and then switching to gauge mode (deciding against it) and then switching back.

Also trying to avoid a situation where the union type becomes a union of a large number of types. While the current situation isn't that overwhelming since the are only 2 I'd set this up anticipating that there would be more in the new future. Maybe I'm overthinking it? 🤔

Reply:
@codeincarnate I don't understand the difference here, the same situation applies on both scenarios, but a union type becomes much stronger typed and less nested. 

I think the subOptions is just very confusing and very badly / untyped in the current model and needs to be fixed

Reply:
> For instance, setting a colored background to "gradient" and then switching to gauge mode (deciding against it) and then switching back.

cellOptions should not have options for not currently set display mode. 

I think misunderstood the model, it's worse than I thought, I thought it was TableCellSubOptionGroup was just a [key: string]: any, but I see it's a record of displaymode enum to the any type. 

```json
 {
        "matcher": {
          "id": "byName",
          "options": "Value"
        },
        "properties": [
          {
            "id": "custom.cellOptions",
            "value": {
                   "displayMode": "gauge",
                   "subOptions": {
                          "gauge": { 
                                  "displayMode": "gradient"    
                            } 
                    }   
             } 
          }
        ]
      }
```

That is unlike any other structure we have for options having a display mode discriminator but then also nested sub options record that duplicates type. Just overly complicated. 

When it could just be 

```json
 {
        "matcher": {
          "id": "byName",
          "options": "Value"
        },
        "properties": [
          {
            "id": "custom.cellOptions",
            "value": {
                   "displayMode": "gauge",                   
                   "gaugeMode": "gradient" ,
             } 
          }
        ]
      }
```

would probably go with "cellType" instead of displayMode but not sure, naming is hard :)

Reply:
OK, agree that this was overly convoluted. I think I have the types in a decent spot now that should work for the cell types we have. I'm a little concerned that it could get crazy as we add more but we can cross that bridge when we get there.

Making corresponding updates for the configuration editor to take into account the new model and will push those all together this evening.

Reply:
Also a per slack conversation with Torkel going to try to cache settings in the editor to deal with times where there is switching between modes and options so that previously set options don't disappear.

Reply:
Alright, took a bit longer than expected but I think I have the types generally worked out. Would be happy for another look before fully diving into the migration.